rmw_opensplice_cpp/src/functions.cpp

```diff
@@ -1126,7 +1126,16 @@ rmw_wait(
       RMW_SET_ERROR_MSG("guard condition handle is null");
       return RMW_RET_ERROR;
     }
-    if (waitset.attach_condition(guard_condition) != DDS::RETCODE_OK) {
+    DDS::ReturnCode_t retcode = waitset.attach_condition(guard_condition);
+    if (retcode == DDS::RETCODE_OUT_OF_RESOURCES) {
+      RMW_SET_ERROR_MSG("failed to attach condition to waitset: out of resources");
+      return RMW_RET_ERROR;
+    }
+    if (retcode == DDS::RETCODE_BAD_PARAMETER) {
+      RMW_SET_ERROR_MSG("failed to attach condition to waitset: invalid condition");
+      return RMW_RET_ERROR;
+    }
+    if (retcode != DDS::RETCODE_OK) {
```

Comment:
Since these condition and code blocks are overlapping I would suggest the following structure to shorten the logic branches:

    if (retcode != DDS::RETCODE_OK) {
      if (retcode == DDS::RETCODE_OUT_OF_RESOURCES) {
        RMW_SET_ERROR_MSG(...);
      } else if (retcode == DDS::RETCODE_BAD_PARAMETER) {
        RMW_SET_ERROR_MSG(...);
      } else {
        RMW_SET_ERROR_MSG(...);
      }
      return RMW_RET_ERROR;
    }